presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -945,6 +953,48 @@ void PrestoServer::addServerPeriodicTasks() {
         ttlCheckInterval,
         "cache_ttl");
   }
+
+  if (isCachePeriodicFullPersistenceEnabled()) {
+    PRESTO_STARTUP_LOG(INFO)
+        << "Initializing cache periodic full persistence task...";
+    auto* cache = velox::cache::AsyncDataCache::getInstance();
+    VELOX_CHECK_NOT_NULL(cache);
+    auto* ssdCache = cache->ssdCache();
+    VELOX_CHECK_NOT_NULL(ssdCache);
+    const auto* systemConfig = SystemConfig::instance();
+    const int64_t cacheFullPersistenceIntervalUs =
+        std::chrono::duration_cast<std::chrono::microseconds>(
+            systemConfig->asyncCacheFullPersistenceInterval())
+            .count();
+    const auto asyncCacheSsdCheckpointGb =
+        systemConfig->asyncCacheSsdCheckpointGb();
+    periodicTaskManager_->addTask(
+        [asyncCacheSsdCheckpointGb, cache, ssdCache]() {
+          try {
+            if (!ssdCache->startWrite()) {
+              return;
+            }
+            LOG(INFO) << "Persisting full cache to SSD...";
+            cache->saveToSsd(true);
+            ssdCache->waitForWriteToFinish();
+
+            if (asyncCacheSsdCheckpointGb == 0) {
+              return;
```

Comment:
Maybe we should add some LOG messages before both returns here to correlate in the log.

Reply:
Good catch. I've moved the log for completion ahead. For checkpoint, we can rely on the logging inside the functions. No need for extra logging here.